processor/src/main/java/org/mapstruct/ap/internal/model/IterableCreation.java

```diff
@@ -69,6 +71,20 @@ public boolean isLoadFactorAdjustment() {
         if ( factoryMethod == null && resultType.getImplementationType() != null ) {
             types.addAll( resultType.getImplementationType().getImportTypes() );
         }
+
+        if ( isEnumSet() ) {
+            types.add( getEnumSetElementType() );
+            // The result type itself is an EnumSet
+            types.add( resultType );
+        }
         return types;
     }
+
+    public Type getEnumSetElementType() {
+        return first( getResultType().determineTypeArguments( Iterable.class ) );
+    }
+
+    public boolean isEnumSet() {
+        return "java.util.EnumSet".equals( resultType.getFullyQualifiedName() );
```

Comment:
What about this?

```suggestion
        return EnumSet.class.getName().equals( resultType.getFullyQualifiedName() );
```

Reply:
I copied this from [`SetterWrapperForCollectionsAndMapsWithNullCheck`](https://github.com/mapstruct/mapstruct/blob/871353fccbf8ed05cfc7dee055f99eb81b0e69e0/processor/src/main/java/org/mapstruct/ap/internal/model/assignment/SetterWrapperForCollectionsAndMapsWithNullCheck.java#L70-L72).

We could do it like your suggestion as well, not big difference I think. I doubt that the `EnumSet` will change it's package anytime soon 😄

Reply:
Totally fine for me